### PR TITLE
fixed ascan category summaries not updated

### LINKED ISSUE #7716: Ascan Category summaries not updated
### Describe the bug

The Active scan dialog Category Threshold and Stength columns are not updated when the "Go" buttons are used.
The underlying rules are however set correctly.
https://groups.google.com/g/zaproxy-users/c/AfVXJpiPPYw/m/m4Jg2KmvAQAJ

### Steps to reproduce the behavior

As per https://groups.google.com/g/zaproxy-users/c/AfVXJpiPPYw/m/m4Jg2KmvAQAJ

### Expected behavior

The Threshold and Strength columns should be updated to reflect the new states.

### Software versions

2.12.0

### Screenshots

See https://groups.google.com/g/zaproxy-users/c/AfVXJpiPPYw/m/m4Jg2KmvAQAJ

### Errors from the zap.log file

_No response_

### Additional context

_No response_

### Would you like to help fix this issue?

- [ ] Yes

### PR DESCRIPTION
**Pull request by aayushhyadav**
Fix #7716

### CODE REVIEW

**Comment by thc202** [comment]
Regresses to #7221.

**Comment by thc202** [comment]
@aayushhyadav ping

**Comment by aayushhyadav** [comment]
> @aayushhyadav ping

?

**Comment by thc202** [comment]
If you plan to do more changes to not introduce the regression.

**Comment by aayushhyadav** [comment]
what changes would be required?

**Comment by thc202** [comment]
That's for you to find out :) (I don't know, I didn't look at it.)

**Comment by aayushhyadav** [comment]
fixed regression issue #7221

**Comment by aayushhyadav** [comment]
@zaproxy/zaproxy-team my changes introduced regression issue (#7221) but I have fixed the issue. Please review my changes and let me know if anything needs to be changed.

**Comment by thc202** [comment]
How are you testing that 7221 is not regressed? I see the main panel shows the expected values but all the other panels do not.

**Comment by aayushhyadav** [comment]
Apologies, I didn't verify the other panels. Thanks for pointing it. I'll look into it.

**Comment by aayushhyadav** [comment]
> How are you testing that 7221 is not regressed? I see the main panel shows the expected values but all the other panels do not.

Resolved.

File: zap/src/main/java/org/zaproxy/zap/extension/ascan/PolicyAllCategoryPanel.java [hunk 1]
```diff
@@ -515,7 +517,9 @@ public void setScanPolicy(ScanPolicy scanPolicy) {
         this.getPolicySelector().setSelectedItem(scanPolicy.getName());
         this.setThreshold(scanPolicy.getDefaultThreshold());
         this.setStrength(scanPolicy.getDefaultStrength());
-        this.getAllCategoryTableModel().setPluginFactory(scanPolicy.getPluginFactory());
+        if (this.policy.equals(scanPolicy)) {
```
Comment (aayushhyadav) [inline zap/src/main/java/org/zaproxy/zap/extension/ascan/PolicyAllCategoryPanel.java:520]: However, I found another issue. Suppose, we update the Default Policy in Scan Policy Manager and restart ZAP, now, if we open the active scan policy panel and try to update the threshold or strength for any specific category, then changes are not reflected on the corresponding category panel. I found that for some reason, `this.policy` and `scanPolicy` do not point to the same object and hence, differ for `PolicyAllCategoryPanel` and `AllCategoryTableModel`. So I added a condition over here.

Comment (thc202) [reply zap/src/main/java/org/zaproxy/zap/extension/ascan/PolicyAllCategoryPanel.java:520]: That's all the same issue: https://github.com/zaproxy/zaproxy/issues/7716#issuecomment-1435108100

**Comment by kingthorin** [comment]
@thc202 is this done?

**Comment by thc202** [comment]
Just rebased, still checking the changes.

File: zap/src/main/java/org/zaproxy/zap/extension/ascan/PolicyAllCategoryPanel.java (hunk 1 above)
Comment (thc202) [reply zap/src/main/java/org/zaproxy/zap/extension/ascan/PolicyAllCategoryPanel.java]: The root cause of the issue was that the combo box was setting another policy. The latest push should fix it, please take a look.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**